File: tests/benchmarks/resolution/fixtures/scala/Service.scala
class UserService(repo: UserRepository) {
  def createUser(id: String, name: String, email: String): Option[User] = {
    val user = User(id, name, email)
    repo.save(user)
    Some(user)
  }

  def getUser(id: String): Option[User] = {
    repo.findById(id)
  }

  def removeUser(id: String): Boolean = {
    repo.delete(id)
  }
}

object ServiceFactory {
  def createService(): UserService = {
    val repo = UserRepository()
    UserService(repo)
  }
}

File: tests/benchmarks/resolution/fixtures/swift/Service.swift
import Foundation

class UserService {
    private let repo: UserRepository

    init(repo: UserRepository) {
        self.repo = repo
    }

    func createUser(id: String, name: String, email: String) -> User? {
        let user = User(id: id, name: name, email: email)
        if !validateUser(user) {
            return nil
        }
        repo.save(user)
        return user
    }

    func getUser(id: String) -> User? {
        return repo.findById(id)
    }

    func removeUser(id: String) -> Bool {
        return repo.delete(id)
    }
}

func createService() -> UserService {
    let repo = UserRepository()
    return UserService(repo: repo)
}
